test/integration/dnspolicy_controller_test.go

```diff
@@ -1330,9 +1370,11 @@ var _ = Describe("DNSPolicy", Ordered, func() {
 				}
 
 				Eventually(func() error {
+					err = k8sClient.Get(ctx, client.ObjectKey{Name: gateway.Name, Namespace: gateway.Namespace}, gateway)
+					Expect(err).NotTo(HaveOccurred())
 					gateway.Spec.Listeners = append(gateway.Spec.Listeners, otherListener)
 					return k8sClient.Update(ctx, gateway)
-				}).Should(BeNil())
+				}, TestTimeoutMedium, TestRetryIntervalMedium).ShouldNot(HaveOccurred())
```

Comment:
This was failing ~60% of the time for me locally. I think the main issue is that the testEnv client is a fake client that updates the whole resource, including the status, when you call update. Needed to get the resource first and update straight away to avoid a conflict.

Reply:
My PR applying these changes as a patch has merged now. If you rebase this it should resolve this problem.